src/dmd/link.d

```diff
@@ -1049,15 +1048,15 @@ version (Windows)
             char[MAX_PATH + 1] dmdpath = void;
             if (GetModuleFileNameA(null, dmdpath.ptr, dmdpath.length) <= MAX_PATH)
             {
-                auto lldpath = FileName.replaceName(dmdpath.ptr, "lld-link.exe");
+                auto lldpath = FileName.replaceName(dmdpath, "lld-link.exe");
                 if (FileName.exists(lldpath))
-                    return lldpath;
+                    return lldpath.ptr;
             }
 
             // search PATH to avoid createProcess preferring "link.exe" from the dmd folder
             Strings* paths = FileName.splitPath(getenv("PATH"));
-            if (auto p = FileName.searchPath(paths, "link.exe", false))
-                return p;
+            if (auto p = FileName.searchPath(paths, "link.exe"[], false))
```

Comment:
Why is this slice needed?

Reply:
Otherwise the call is ambiguous, since we have a `const(char)*` and a `const(char)[]` overload, and manifest constant convert to both.
In a later PR (because this is already quite big) I'll remove the `const(char)*` overload

Reply:
Ah, ok.